shopBuddy/shopBuddy/ScanBarcodeViewController.m

```diff
@@ -0,0 +1,107 @@
+//
+//  ScanBarcodeViewController.m
+//  shopBuddy
+//
+//  Created by Rachna Gupta on 7/14/22.
+//
+
+#import "ScanBarcodeViewController.h"
+#import "AVFoundation/AVFoundation.h"
+#import "ItemDetailViewController.h"
+
+@interface ScanBarcodeViewController ()
+@property (weak, nonatomic) IBOutlet UIView *preview;
+@property (weak, nonatomic) IBOutlet UIBarButtonItem *scanButton;
+@property (weak, nonatomic) IBOutlet UIButton *lookupButton;
+@property NSString *barcode;
+
+@end
+@implementation ScanBarcodeViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    _isScanning = NO;
+    _captureSession = nil;
+    [_lookupButton setTitle:@" " forState:UIControlStateNormal];
+    
+}
+//changes the boolean property if the stop/start button is pressed
+- (IBAction)startStopScanning:(id)sender {
+    if (!_isScanning) {
+            if ([self startScanning]) {
+                [_scanButton setTitle:@"Stop Scanning"];
+            }
+        }
+        else{
+            [self stopScanning];
+            [_scanButton setTitle:@"Start Scanning"];
+        }
+        
+        _isScanning = !_isScanning;
+    }
+//starts scanning for barcodes (metadata objects) and runs the capturesession
+- (BOOL)startScanning {
+    NSError *error;
+ 
+    AVCaptureDevice *captureDevice = [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo];
+    AVCaptureDeviceInput *input = [AVCaptureDeviceInput deviceInputWithDevice:captureDevice error:&error];
+     
+        if (!input) {
+            NSLog(@"%@", [error localizedDescription]);
+            return NO;
+        }
+    _captureSession = [[AVCaptureSession alloc] init];
+        [_captureSession addInput:input];
+    AVCaptureMetadataOutput *captureMetadataOutput = [[AVCaptureMetadataOutput alloc] init];
+        [_captureSession addOutput:captureMetadataOutput];
+    dispatch_queue_t dispatchQueue;
+        dispatchQueue = dispatch_queue_create("myQueue", NULL);
+        [captureMetadataOutput setMetadataObjectsDelegate:self queue:dispatchQueue];
+        [captureMetadataOutput setMetadataObjectTypes:[NSArray arrayWithObject:AVMetadataObjectTypeEAN13Code]];
+        _videoPreviewLayer = [[AVCaptureVideoPreviewLayer alloc] initWithSession:_captureSession];
+        [_videoPreviewLayer setVideoGravity:AVLayerVideoGravityResizeAspectFill];
+        [_videoPreviewLayer setFrame:_preview.layer.bounds];
+        [_preview.layer addSublayer:_videoPreviewLayer];
+    [_captureSession startRunning];
+    return YES;
+    
+}
+//run when an object is captured (a barcode is scanned)
+-(void)captureOutput:(AVCaptureOutput *)captureOutput didOutputMetadataObjects:(NSArray *)metadataObjects fromConnection:(AVCaptureConnection *)connection{
+    if (metadataObjects != nil && [metadataObjects count] > 0) {
+        AVMetadataMachineReadableCodeObject *metadataObj = [metadataObjects objectAtIndex:0];
```

Comment:
`AVMetadataMachineReadableCodeObject *const metadataObj = ...` declare as const if value is not expected to change.

Reply:
not resolved